plugin/forward/forward.go

```diff
@@ -65,24 +66,36 @@ func (f *Forward) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg
 	}
 
 	fails := 0
-	var span, child ot.Span
-	var upstreamErr error
-	span = ot.SpanFromContext(ctx)
+	i := 0
+	returnErr := errNoHealthy
+	span := ot.SpanFromContext(ctx)
+	proxyList := f.list()
+	deadline := time.Now().Add(queryTimeout)
+
+	for time.Now().Before(deadline) {
+		if i >= len(proxyList) {
+			if i == fails {
+				// All upstream proxies are dead, assume healtcheck is completely broken and randomly
+				// select an upstream to connect to.
+				i = rand.Intn(len(proxyList))
```

Comment:
Just leave the original code here?

Reply:
the original code was more expensive from performance point of view, so I changed it